extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/CertificateConfig.java

```diff
@@ -75,7 +75,14 @@
     public String keyStorePassword;
 
     /**
-     * An optional trust store which holds the certificate information of the certificates to trust
+     * An optional parameter to select a specific key in the key store. If no such alias is defined and there are multiple keys
+     * in the store, behavior is undefined.
```

Comment:
Probably worth mentioning SNI.

Reply:
I found it a bit dangerous to ignore the alias if one is set and none match. Anything we can do to reject the configuration in this case?

Reply:
+1, if the user specifies the alias then it has to match or fail

Reply:
I wonder if it should be detected at the Vert.x level though ? Perhaps it is already detected ?

Reply:
I did test a non-matching alias and was surprised myself that vert.x is not bailing out with an error.
I do also think that this is not user-friendly but do we really want to handle this in Quarkus? AFAICS we would need to "open" the store ourself to check it.

Reply:
@famod I agree it is not a Quarkus-level problem since Vert.x is managing the stores. Can you please open a Vert.x issue for the error be reported if no matching key exists ?
@cescoffier Do you agree ?

Reply:
@sberyozkin
>  Can you please open a Vert.x issue for the error be reported if no matching key exists ?

I'll try. 😉 

@cescoffier
> Probably worth mentioning SNI.

I have only a vague understanding of SNI. Do you have a suggestion? Maybe something like:
> If no such alias is defined and there are multiple keys in the store, behavior is undefined (unless SNI is used).

Reply:
Yes something like this. SNI allows having multiple certificates and the server name allows using the right one. 
So something like: 

```
When SNI is disabled, if the key store contains multiple keys and no alias is specified, the behavior is undefined.
```

Reply:
@cescoffier ok thanks, will use your suggestion.

I also suppose your are ok with having alias checking implemented in a later vertx release, as suggested by @sberyozkin.

Reply:
Yes.